### ISSUE TITLE
The generated name is the same as the package name

### ISSUE THREAD

**Issue by godcong** [facebook/ent#1225]
I created a table using `ent init Config`,
after i generate the code use `ent generate ./ent/schema`.
The program was compiled with error

error code:
```
ent\client.go:13:2: config redeclared in this block
	previous declaration at ent\config.go:14:6
```

- [x ] The issue is present in the latest release.
- [x ] I have searched the [issues](https://github.com/facebook/ent/issues) of this repository and believe that this is not a duplicate.

## Steps to Reproduce 🕹

Steps:

1. ent init Config
2. ent generate ./ent/schema
3. coding a main function
4. build and run it

## Your Environment 🌎

| Tech        | Version |
| ----------- | ------- |
| Go          | 1.15.6  |
| Ent         | 0.5.4  |
| Database    | sqlite3|
| Driver      | https://github.com/mattn/go-sqlite3        |

**Comment by a8m** [comment]
Hey @godcong and thanks for reporting this issue.
Yes, `config` is a predeclared identifier in the generated `ent` package, and we need to add a check for this before running codegen. You're welcome to contribute this change and add the `Config` name [here](https://github.com/facebook/ent/blob/master/entc/gen/type.go#L1343-L1367).

**Comment by godcong** [comment]
I think the problem of duplicate names can be solved by adding an alias name before import

like:
```
import(
"project/ent/config"
)

=>

import(
entconfig "project/ent/config"
)
```

**Comment by napei** [comment]
> and we need to add a check for this before running codegen.

But this should already be checked at time of running `ent init` thanks to #1110 so just adding this to the list of checks against `globalIdent` should sort it out. However, `ValidSchemaName` should really check the `privateField` array as well, which already contains `config`. 

https://github.com/facebook/ent/blob/0f6c71d46b2f0c868c3e88bd9304ca000f26620d/entc/gen/type.go#L685-L698

Adding this will also protect it at codegen time as well. I think.

https://github.com/facebook/ent/blob/dd4792f5b30bdd2db0d9a593a977a54cb3f0c1ce/entc/gen/type.go#L189-L191

Apologies for the interruption, just noticed this as I recognised it as check the check I added :) Happy to make a PR for this.
